### DISCUSSION TITLE
Does it support mpv.conf file and scripts?

### DISCUSSION THREAD

**Discussion by mysteryx93** [smplayer-dev/smplayer#364] [Q&A]
MPV supports configuration file and powerful scripting engine. Can those be used in SMPlayer, or is the only option to fill the Options field in Advanced |MPV ?

Basically -- do we have to choose between a decent UI *or* scripts and conf file?

One thing that can't be done with the Options field is adjusting settings based on video size.

**Comment by smplayer-dev** [comment]
I'll try to fix it.

**Comment by mysteryx93** [comment]
In the future I may have a use-case where I'll control MPV or SMPlayer from another app, and would rather configure it by passing it a .conf file. Allowing for different configuration than when they use the player on its own. Loading arbitrary .conf files would be a useful feature; but perhaps that should be a feature of MPV itself, if it doesn't have it already.

**Comment by smplayer-dev** [comment]
The problem should be fixed now: https://github.com/smplayer-dev/smplayer/commit/2d44054ce7c3bba6dcae2df127b8f5e82e5ba549

**Comment by mysteryx93** [comment]
ah so now it will load the standard MPV config file? How does `--noconfig` work from libmpv? There may not even be a standard MPV installation on the system.

But what I'm saying, loading specific config files, I see it's fully supported with `--config-dir`, `--include`, `--script` and several other keys. Perhaps it would be wiser to implement that in a way that is not dependent nor altered by a stand-alone MPV installation.

You could easily add a .conf file for SMPlayer, and a tab to edit it directly, plus something to easily support scripts.

To add this support properly, would have to sit down and take the time to design it the right way. MPV itself is very flexible and makes all the options available.

My 2 cents.

**Comment by mysteryx93** [comment]
or; leave it as it is. It is possible to use config files and scripts by setting `config-dir` or other params; it's just not obvious and almost nobody knows it.

How about... simply setting `config-dir` by default to the path where SMPlayer stores its config? and if we manually set --config-dir, don't override or duplicate it. And have a config file and scripts folder so that people know they can use it. Wouldn't require UI changes.

**Comment by polarathene** [comment]
One related feature I noticed SMPlayer didn't cover in the UI was config profiles. MPV config profiles can presumably be adjusted via specifying a default via option and overriding if needed via CLI?
